run_it_diffeq_expl.jl

```diff
@@ -104,36 +103,40 @@ function make_ode()
         vc     .=  h .* (0.91 .* H .- ϕ).^3
 
         dhdt   .= (Σ .* vo .- Γ .* vc)
-        dϕdt   .= (.- (diff(flux_x[:,2:end-1],dims=1) ./ dx .+ diff(flux_y[2:end-1,:],dims=2) ./ dy) .- inn(dhdt) .+ Λ) ./ e_v
+        div_q  .= (diff(flux_x[:,2:end-1],dims=1) ./ dx .+ diff(flux_y[2:end-1,:],dims=2) ./ dy)
+        dϕdt   .= (.- div_q .- inn(dhdt) .+ Λ) ./ e_v
+        if use_masscons_for_h
+            # note this is 100% identical (modulus floating point errors) to use_masscons_for_h==false...
+            inn(dhdt) .= .-e_v.*dϕdt .- div_q .+ Λ
```

Comment:
This is the mass-cons scheme of B&P.  However, note that it is 100% the same as doing the ODE.  I'm not sure whether in their FV-scheme there is a difference.

Reply:
Here the `dϕdt` in L110 is the updated one, shouldn't it refer to the one from the previous time step, if it is a purely explicit scheme?

Reply:
No.  Explicit means that it doesn't use information from the "future" (which can only be gained by solving non-lin eqs).

Reply:
Ah sorry I did a thinking mistake, I was just a bit confused that `dhdt` and `dϕdt` are using each other but then I saw that `dhdt` is updated twice. Clear now, thanks.